web/src/activity_ui.ts

```diff
@@ -152,18 +150,6 @@ export function initialize(opts: {narrow_by_email: (email: string) => void}): vo
     build_user_sidebar();
 
     buddy_list.start_scroll_handler();
-
-    function get_full_presence_list_update(): void {
-        activity.send_presence_to_server(redraw);
-    }
-
-    /* Time between keep-alive pings */
-    const active_ping_interval_ms = realm.server_presence_ping_interval_seconds * 1000;
-    util.call_function_periodically(get_full_presence_list_update, active_ping_interval_ms);
-
-    // Let the server know we're here, but do not pass
-    // redraw, since we just got all this info in page_params.
-    activity.send_presence_to_server();
```

Comment:
Hmm, this isn't the way I'd expect us to cut a dependency cycle here; what is the cycle in question?

Reply:
I recall it was `activity_ui` imported by both `message_events -> recent_view_ui -> views_util` and `activity`

Reply:
Let me recheck.

Reply:
```bash
eslint    | 
eslint    | /home/aman/zulip/web/src/activity.ts
eslint    |   9:1  error  Dependency cycle via ./recent_view_ui:23=>./views_util:48=>./activity_ui:4  import/no-cycle
eslint    | 
eslint    | ✖ 1 problem (1 error, 0 warnings)
```

Reply:
So, `message_events` imports `activity` and also imports `./recent_view_ui:23=>./views_util:48=>./activity_ui->actvity`.

Reply:
See https://github.com/zulip/zulip/pull/32038#discussion_r1807005741